code/mission/missioncampaign.cpp

```diff
@@ -1883,13 +1883,21 @@ void mission_campaign_jump_to_mission(char *name)
 
 	// search for our mission
 	for (i = 0; i < Campaign.num_missions; i++) {
-		if ((Campaign.missions[i].name != NULL) && !stricmp(Campaign.missions[i].name, dest_name)) {
-			mission_num = i;
-			break;
+		if (no_skip) {							// Prevents CMISSION_FLAG_SKIPPED from being set on missions
+			if ((Campaign.missions[i].name != NULL) && !stricmp(Campaign.missions[i].name, dest_name)) {
+				mission_num = i;
+				break;
+			}
 		} else {
-			Campaign.missions[i].flags |= CMISSION_FLAG_SKIPPED;
-			Campaign.num_missions_completed = i;
+			if ((Campaign.missions[i].name != NULL) && !stricmp(Campaign.missions[i].name, dest_name)) {
+				mission_num = i;
+				break;
+			} else {
```

Comment:
This could still be simplified a bit. If you use `} else if (!no_skip) {` you don't need to duplicate the three lines above.